[docs][PLAT-6522] Docs for YBA Kubernetes Multi-Cluster Services (#16536)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-platform/install-yugabyte-platform/prerequisites/kubernetes.md

      <i class="fa-solid fa-cloud"></i>Default</a>
  </li>

  <li>
    <a href="../kubernetes/" class="nav-link active">
      <i class="fa-regular fa-dharmachakra" aria-hidden="true"></i>Kubernetes</a>
  </li>

</ul>

YugabyteDB Anywhere first needs to be installed on a Kubernetes cluster, and then you configure YugabyteDB Anywhere to work in your Kubernetes cluster, on-premises private cloud, or in a public cloud environment.

## Hardware requirements

A Kubernetes node is expected to meet the following requirements: 

- 5 cores (minimum) or 8 cores (recommended)
- 15 GB RAM (minimum)
- 250 GB SSD disk (minimum)
- 64-bit CPU architecture

## Prepare the cluster

The YugabyteDB Anywhere Helm chart has been tested using the following software versions:

- Kubernetes 1.22
- Helm 3.10

Before installing YugabyteDB Anywhere, verify that you have the following:

- A Kubernetes cluster with nodes configured according to the [hardware requirements](#hardware-requirements).
- A client environment with the kubectl and Helm command-line tools configured with a service account or user that has admin access to a  single namespace on the subject Kubernetes cluster.

> - A Kubernetes secret obtained from [Yugabyte](https://www.yugabyte.com/platform/#request-trial-form).

Reviewer: ```suggestion
- A Kubernetes secret obtained from {{% support-platform %}}.
```

Author: Oh! Nice.

- The nodes can pull container images from the [quay.io](https://quay.io/) container registry. If the nodes cannot do this, you need to prepare these images in your internal registry by following instructions provided in [Pull and push YugabyteDB Docker images to private container registry](../../prepare-environment/kubernetes#pull-and-push-yugabytedb-docker-images-to-private-container-registry). 
- Core dumps are enabled and configured on the underlying Kubernetes node. For details, see [Specify ulimit and remember the location of core dumps](../../prepare-environment/kubernetes#specify-ulimit-and-remember-the-location-of-core-dumps).
- You have the kube-state-metrics add-on version 1.9 in your Kubernetes cluster. For more information, see [Install kube-state-metrics](../../prepare-environment/kubernetes#install-kube-state-metrics).
- A load balancer controller is available in your Kubernetes cluster.
- A StorageClass is available with the SSD and `WaitForFirstConsumer` preferably set to `allowVolumeExpansion`. For more information, see [Configure storage class](../../prepare-environment/kubernetes/#configure-storage-class).

## Multi cluster Kubernetes environment

> When you want to create a multi-region YugabyteDB universe or two universes from different regions replicating using xCluster, you need to use one Kubernetes cluster per region. Following things are required for YugabyteDB Anywhere to work across multiple Kubernetes clusters:

Reviewer 2: In the common case, the following pre-requisites are required ...

> - Pod IP address connectivity should be present between the clusters. Each pod and service should have an unique IP address across the clusters (non-overlapping addresses).

Reviewer: ```suggestion
- Pod IP address connectivity should be present between the clusters. Each pod and service should have a unique IP address across the clusters (non-overlapping addresses).
```

> - There should be DNS connectivity between the clusters. ClusterIP and headless service FQDNs (including the individual pod FQDNs) exposed in one Kubernetes cluster should be resolvable in all the other Kubernetes clusters.

Reviewer: Cluster IP?

Author: ClusterIP is a type of service in Kubernetes i.e. it is a keyword.

> - DNS connectivity between the clusters. ClusterIP and headless service pod FQDNs should be resolvable across the Kubernetes clusters. For example, `yb-master-0.yb-masters.ns.svc.cluster.local` and `yb-tserver-service.ns.svc.cluster.local` should get resolved.

Reviewer 2: The For example part is not needed maybe. 

Cluster IP and headless service FQDNs exposed in one kubernetes cluster should be resolvable in all the other kubernetes clusters.

Author: Modified it a bit. Only confusion there was with the individual pod addresses, have clarified that in the sentence now.

> - YugabyteDB Anywhere should have access to the control plane of all the Kubernetes clusters to perform Helm and kubectl operations against them. It should be installed on one of the connected clusters.

Reviewer 2: "to perform helm and kubectl operations against them" might not be necessary. But we can say "should have access to the control plane, typically via a KUBECONFIG file for a service account."

> Alternatively, you can setup [Multi-Cluster Services API](https://git.k8s.io/enhancements/keps/sig-multicluster/1645-multi-cluster-services-api) (MCS). For more details on the setup, see [Configure Kubernetes multi-cluster environment](../../../configure-yugabyte-platform/set-up-cloud-provider/kubernetes#configure-kubernetes-multi-cluster-environment). Note that MCS support in YugabyteDB Anywhere is still in beta.

Reviewer: ```suggestion
Alternatively, you can set up [Multi-Cluster Services API](https://git.k8s.io/enhancements/keps/sig-multicluster/1645-multi-cluster-services-api) (MCS). For more details on the setup, see [Configure Kubernetes multi-cluster environment](../../../configure-yugabyte-platform/set-up-cloud-provider/kubernetes#configure-kubernetes-multi-cluster-environment). Note that MCS support in YugabyteDB Anywhere is currently in [Beta](/preview/faq/general/#what-is-the-definition-of-the-beta-feature-tag).
```

> If your cluster setup does not satisfy above requirements, you can setup [Multi-Cluster Services API](https://git.k8s.io/enhancements/keps/sig-multicluster/1645-multi-cluster-services-api) (MCS). For more details on the setup, see [Configure Multi-Cluster Services](../../../configure-yugabyte-platform/set-up-cloud-provider/kubernetes#configure-multi-cluster-services). Note that MCS support in YugabyteDB Anywhere is still in beta.

Reviewer 2: Alternately, you can setup ..
